plugins/package-managers/python/src/main/kotlin/Poetry.kt

```diff
@@ -73,14 +76,17 @@ class Poetry(
                 version = VersionControlSystem.getCloneInfo(definitionFile.parentFile).revision
             ),
             definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
-            scopeDependencies = setOf(Scope("install", result.resolvedDependenciesGraph.toPackageReferences())),
+            scopeDependencies = setOf(
+                Scope("install", dependenciesResult.resolvedDependenciesGraph.toPackageReferences()),
+                Scope("installDev", devDependenciesResult.resolvedDependenciesGraph.toPackageReferences())
+            ),
             vcsProcessed = processProjectVcs(definitionFile.parentFile)
         )
 
         return listOf(ProjectAnalyzerResult(project, packages))
     }
 
-    private fun inspectLockfile(lockfile: File): PythonInspector.Result {
+    private fun inspectLockfile(lockfile: File,  devDependencies: Boolean): PythonInspector.Result {
```

Comment:
I'd propose to generalize this further and take the dependency group name as a string instead, and pass that to `--only=`. The same names should be used for the scopes then. As a preparation, it might make sense to first rename "install" to "main" to reduce the diff in the expected output. So in total, we should have two scopes, "main" and "dev".